test/parallel/test-abortcontroller.js

```diff
@@ -231,6 +233,30 @@ const { setTimeout: sleep } = require('timers/promises');
   AbortSignal.timeout(1_200_000);
 }
 
+{
+  (async () => {
+    // Test AbortSignal timeout can be GCed when no listeners exist
+    let ref;
+    {
+      function lis() {
+
+      }
+      ref = new globalThis.WeakRef(AbortSignal.timeout(1_200_000));
+      ref.deref().addEventListener('abort', lis);
+      aborted(ref.deref(), {});
+
+      await sleep(10);
```

Comment:
Why is this sleep required?

Reply:
https://github.com/nodejs/node/pull/48952#discussion_r1278601187